### PR TITLE
Update to Scala 2.13

### PR DESCRIPTION
**Pull request by poslegm**
Hello!

I tried to update paiges to 2.13 and bring publishing closer. 

Current problems:

1. A lot of warnings about `Stream` deprecation (I think that it can be fixed after first `_2.13` release :pray:)
2. Weird errors in tests:

```
[error] Referring to non-existent method org.typelevel.paiges.PaigesTest.assertionsHelper()org.scalatest.Assertions$AssertionsHelper
[error]   called from org.typelevel.paiges.PaigesTest.$$anonfun$new$9()org.scalatest.compatible.Assertion
[error]   called from org.typelevel.paiges.PaigesTest.<init>()
[error]   called from org.typelevel.paiges.PaigesTest.<clinit>()
[error]   called from core module analyzer
[error] involving instantiated classes:
[error]   org.typelevel.paiges.PaigesTest
```

Looks like scalatest version mismatched, any ideas about it?

#152

### CODE REVIEW

**Comment by johnynek** [comment]
Huh. No I don’t know the cause of those errors. I wonder if we are keeping a cache between builds and that could be related? Can you reproduce locally?

Also, I’d double check the latest version of scalatest or consider only using scalacheck, which might be a bigger port.

**Comment by poslegm** [comment]
Yes, it reproducible locally. I suspect that problem may be in the inconsistency scalatest versions in `build.sbt` and last discipline release.

File: build.sbt [hunk 1]
```diff
@@ -90,8 +90,10 @@ lazy val cats = crossProject(JSPlatform, JVMPlatform)
     name := "paiges-cats",
     moduleName := "paiges-cats",
     libraryDependencies ++= Seq(
-      "org.typelevel" %%% "cats-core" % "1.6.1",
-      "org.typelevel" %%% "cats-laws" % "1.6.1" % Test),
+      "org.typelevel" %%% "cats-core" % "2.0.0-M4",
+      "org.typelevel" %%% "cats-laws" % "2.0.0-M4" % Test,
+      "org.typelevel" %%% "discipline-core" % "0.12.0-M3" % Test,
```
Comment (kailuowang) [inline build.sbt]: it's used by cats-laws, so you don't really need this.

File: build.sbt [hunk 2]
```diff
@@ -63,7 +63,7 @@ lazy val core = crossProject(JSPlatform, JVMPlatform, NativePlatform)
   .jvmSettings(commonJvmSettings)
   .platformsSettings(JVMPlatform, JSPlatform)(
     libraryDependencies ++= Seq(
-      "org.scalatest" %%% "scalatest" % "3.0.6-SNAP5" % Test,
+      "org.scalatest" %%% "scalatest" % "3.0.8" % Test,
```
Comment (kailuowang) [inline build.sbt]: You need to update to `3.1.0-SNAP13` (which will likely require some changes to the usage code), 3.1.0 is used by `discipline-scalatest` and is not compatible with 3.0.8

File: build.sbt (hunk 1 above)
Comment (poslegm) [reply build.sbt]: Removed, thank you

File: build.sbt (hunk 2 above)
Comment (poslegm) [reply build.sbt]: Yes, it works! Thank you!

File: core/src/test/scala/org/typelevel/paiges/PaigesTest.scala [hunk 3]
```diff
@@ -1,7 +1,7 @@
 package org.typelevel.paiges
 
 import scala.util.Random
-import org.scalatest.FunSuite
+import org.scalatest.funsuite.AnyFunSuite
```
Comment (johnynek) [inline core/src/test/scala/org/typelevel/paiges/PaigesTest.scala]: why is this change needed?

Comment (poslegm) [reply core/src/test/scala/org/typelevel/paiges/PaigesTest.scala]: Because of deprecation warning in scalatest https://github.com/scalatest/scalatest/blob/b54539cd8057b2c8207e5390c2aef83b95756538/scalatest/src/main/scala/org/scalatest/FunSuite.scala#L22

Comment (johnynek) [reply core/src/test/scala/org/typelevel/paiges/PaigesTest.scala]: that's a bummer. `AnyFunSuite` seems like a worse API and I like using FunSuite.
